src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -116,6 +117,7 @@
   private final AtomicBoolean isCopying = new AtomicBoolean();
 
   private MongoSourceConfig sourceConfig;
+  private Map<String, Object> partitionMap;
```

Comment:
Is this field just a cache for efficiency?  If so, it doesn't seem necessary, as creating the value is not resource-intensive.

Reply:
Its cached because it can be called thousands of times - one per source record produced / change stream event